File: js/11.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (h) {
    let getArea = (i, j) => (j - i) * Math.min(h[i], h[j])
    let i = 0, j = h.length - 1, ret = 0
    while (i < j) {
        ret = Math.max(ret, getArea(i, j))
        if (h[i] < h[j]) i++
        else j--
    }
    return ret
};

File: cpp/11.cpp
class Solution {
 public:
  inline int getArea(int i, int j, vector<int>& h) {
    return (j - i) * min(h[i], h[j]);
  }
  int maxArea(vector<int>& h) {
    int n = h.size();
    int i = 0, j = n - 1, ret = 0;
    while (i < j) {
      int t = getArea(i, j, h);
      ret = max(ret, t);
      if (h[i] < h[j]) i++;
      else j--;
    }
    return ret;
  }
};